examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,276 @@
+train_pipe = train_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+train_pipe = train_pipe.shuffle(buffer_size).batch(batch_size).repeat()
+
+
+val_pipe = tf.data.Dataset.from_tensor_slices((val_source, val_targets))
+val_pipe = val_pipe.map(data_preprocess, tf.data.AUTOTUNE).cache()
+val_pipe = val_pipe.batch(batch_size).repeat()
+
+
+train_step = len(train_source) // batch_size
+val_step = len(val_source) // batch_size
+
+fig, ax = plt.subplots(1, 2, figsize=(16, 8))
+ax = ax.flatten()
+for i, f in enumerate(val_pipe.take(1)):
+    noise, clean = f
+    noise = noise[i].numpy() + 1.0
+    noise = noise * 127.5
+    clean = clean[i].numpy() + 1.0
+    clean = clean * 127.5
+    ax[0].imshow(noise.astype(np.int))
+    ax[0].set_xlabel("source")
+    ax[1].imshow(clean.astype(np.int))
+    ax[1].set_xlabel("target")
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
+    base_model = resnet_v2.ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(352, 608, 3)
+    )
+    d0 = base_model.get_layer("input_1").output
+    d1 = base_model.get_layer("conv1_conv").output
+    d2 = base_model.get_layer("conv2_block3_1_relu").output
+    d3 = base_model.get_layer("conv3_block4_1_relu").output
+
+    decode1 = upscale_unit(previous_input=d3, current_input=d2, target_channels=128)
+    decode2 = upscale_unit(previous_input=decode1, current_input=d1, target_channels=64)
+    decode3 = upscale_unit(previous_input=decode2, current_input=d0, target_channels=32)
+
+    out_conv = keras.layers.Conv2D(3, 5, padding="same", activation="sigmoid")(decode3)
+    denoiser_model = keras.Model(base_model.input, out_conv, name="denoiser")
+    return denoiser_model
+
+
+"""
+## Training
+"""
+
+denoiser_net = denoiser()
```

Comment:
@anish9 I think after building the model, it's nice to show it's summary or graph (plot_model).